atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -43,14 +45,62 @@ def __init__(self, tasker, workflow, registries):
 
         self.registries = deepcopy(registries)
 
+    def need_skopeo_push(self):
+        if len(self.workflow.exported_image_sequence) > 0:
+            last_image = self.workflow.exported_image_sequence[-1]
+            if last_image['type'] == IMAGE_TYPE_OCI or last_image['type'] == IMAGE_TYPE_OCI_TAR:
+                return True
+
+        return False
+
+    def push_with_skopeo(self, registry_image, insecure, docker_push_secret):
+        # If the last image has type OCI_TAR, then hunt back and find the
+        # the untarred version, since skopeo only supports OCI's as an
+        # untarred directory
+        index = 1
+        while index <= len(self.workflow.exported_image_sequence):
+            image = self.workflow.exported_image_sequence[-index]
+            if image['type'] != IMAGE_TYPE_OCI_TAR:
+                break
+            index += 1
+        assert image is not None
+
+        cmd = ['skopeo', 'copy']
+        if docker_push_secret is not None:
+            dockercfg = Dockercfg(docker_push_secret)
+            credentials = dockercfg.get_credentials(registry_image.registry)
+            username = credentials['username']
+            password = credentials['password']
+
+            cmd.append('--dest-creds=' + username + ':' + password)
+
+        if insecure:
+            cmd.append('--dest-tls-verify=false')
+
+        if image['type'] == IMAGE_TYPE_OCI:
+            source_img = 'oci:' + image['path'] + '::' + image['ref-name']
+        elif image['type'] == IMAGE_TYPE_DOCKER_ARCHIVE:
+            source_img = 'docker-archive://' + image['path']
+        else:
+            raise RuntimeError("Attempt to push unsupported image type %s with skopeo",
+                               image['type'])
+
+        dest_img = 'docker://' + registry_image.to_str()
+
+        cmd += [source_img, dest_img]
+
+        self.log.info("Calling: %s", ' '.join(cmd))
```

Comment:
Thanks for catching this - had noticed that, but didn't do anything hoping to find some better way to send auth to skopeo. If I don't log, and make sure that failure of skopeo doesn't log, do you think this is good enough (probably don't need to worry about ps on the build system)? The only alternative I can find is to create a FAKE_HOMEDIR/.docker/config.json and point to that with $HOME when calling skopeo, which is a lot of work, and somewhat fragile w.r.t. to the exact handling of config.json.